text/0000-incremental-compilation.md

```diff
@@ -0,0 +1,648 @@
+- Feature Name: incremental-compilation
+- Start Date: 2015-08-04
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Enable the compiler to cache incremental workproducts.
+
+# Motivation
+
+The goal of incremental compilation is, naturally, to improve build
+times when making small edits. Any reader who has never felt the need
+for such a feature is strongly encouraged to attempt hacking on the
+compiler or servo sometime (naturally, all readers are so encouraged,
+regardless of their opinion on the need for incremental compilation).
+
+## Basic usage
+
+The basic usage will be that one enables incremental compilation using
+a compiler flag like `-C incremental-compilation=TMPDIR`. The `TMPDIR`
+directory is intended to be an empty directory that the compiler can
+use to store intermediate by-products; the compiler will automatically
+"GC" this directory, deleting older files that are no longer relevant
+and creating new ones.
```

Comment:
When does this GC happen? Will it be sort of like Git where GCs can potentially happen whenever you type any command?